modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/wstep/impl/CertificateEnrollmentServiceImpl.java

```diff
@@ -126,101 +118,54 @@ public void requestSecurityToken(String tokenType, String requestType,
-            }
-            if (log.isDebugEnabled()) {
-                log.debug("Received CSR from Device:" + binarySecurityToken);
-            }
-            String wapProvisioningFilePath = wapProvisioningFile.getPath();
-            RequestSecurityTokenResponse requestSecurityTokenResponse =
-                    new RequestSecurityTokenResponse();
-            requestSecurityTokenResponse.setTokenType(PluginConstants.CertificateEnrolment.TOKEN_TYPE);
-            String encodedWap;
-            try {
-                encodedWap = prepareWapProvisioningXML(binarySecurityToken, certPropertyList,
-                        wapProvisioningFilePath, headerBinarySecurityToken);
-                RequestedSecurityToken requestedSecurityToken = new RequestedSecurityToken();
-                BinarySecurityToken binarySecToken = new BinarySecurityToken();
-                binarySecToken.setValueType(PluginConstants.CertificateEnrolment.VALUE_TYPE);
-                binarySecToken.setEncodingType(PluginConstants.CertificateEnrolment.ENCODING_TYPE);
-                binarySecToken.setToken(encodedWap);
-                requestedSecurityToken.setBinarySecurityToken(binarySecToken);
-                requestSecurityTokenResponse.setRequestedSecurityToken(requestedSecurityToken);
-                requestSecurityTokenResponse.setRequestID(REQUEST_ID);
-                response.value = requestSecurityTokenResponse;
-            }
-            //Generic exception is caught here as there is no need of taking different actions for
-            //different exceptions.
-            catch (Exception e) {
-                String msg = "Wap provisioning file couldn't be prepared.";
-                log.error(msg, e);
-                throw new WindowsDeviceEnrolmentException(msg, e);
-            }
-        } catch (DeviceManagementException e) {
-            String msg = "Error occured in while getting tenant configuration.";
+            encodedWap = prepareWapProvisioningXML(binarySecurityToken,
+                    wapProvisioningFilePath, headerBinarySecurityToken);
+            RequestedSecurityToken requestedSecurityToken = new RequestedSecurityToken();
+            BinarySecurityToken binarySecToken = new BinarySecurityToken();
+            binarySecToken.setValueType(PluginConstants.CertificateEnrolment.VALUE_TYPE);
+            binarySecToken.setEncodingType(PluginConstants.CertificateEnrolment.ENCODING_TYPE);
+            binarySecToken.setToken(encodedWap);
+            requestedSecurityToken.setBinarySecurityToken(binarySecToken);
+            requestSecurityTokenResponse.setRequestedSecurityToken(requestedSecurityToken);
+            requestSecurityTokenResponse.setRequestID(REQUEST_ID);
+            response.value = requestSecurityTokenResponse;
+        }
+        //Generic exception is caught here as there is no need of taking different actions for
+        //different exceptions.
+        catch (Exception e) {
```

Comment:
As a best practice use multi-catch statements for better robustness of the code, instead of catching generic exceptions.